compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -397,9 +397,20 @@ object ProtoTypes {
         tt.withType(new TypeVar(tl.paramRefs(n), state, tt, ctx.owner))
       }
 
-    val added =
-      if (state.constraint contains tl) tl.newLikeThis(tl.paramNames, tl.paramInfos, tl.resultType)
+    /** Ensure that `tl` is not already in constraint, make a copy of necessary */
+    def ensureFresh(tl: TypeLambda): TypeLambda =
+      if (state.constraint contains tl) {
+      	var paramInfos = tl.paramInfos
+      	if (tl.isInstanceOf[HKLambda]) {
+      	  // HKLambdas care hash-consed, need to create an artificial difference by adding
```

Comment:
typo: care -> are

Reply:
Would something break if we did `tl.clone()` instead to get a different instance?

Reply:
Yes, they would be assumed as `equals`, so e.g. pendingSubtypes would not distinguish between them.